eitprocessing/binreader/sequence.py

```diff
@@ -96,8 +96,11 @@ def __eq__(self, other) -> bool:
                 return False
             if not np.all(np.equal(self_attr, other_attr)):
                 return False
-
-        return True
+        try:
+            self.check_equivalence(self, other)
+            return True
+        except (TypeError, ValueError, AttributeError):
+            return False
```

Comment:
yeah, I figured that would be the case, but wasn't sure exactly what so thought it would be "extra safe" and not too expensive to just run the `check_equivalence`.
Good idea to switch the order.